frontend/model/state.js

```diff
@@ -108,37 +127,59 @@ const mutations = {
-    const prevOthers = state.chatRoomUnread[chatRoomId].others
-
+    // eslint-disable-next-line no-lone-blocks
+    {
+      // TODO: need to remove this block after we release 0.2.*
+      if (!state.chatRoomUnread[chatRoomId].messages) {
+        Vue.set(
+          state.chatRoomUnread[chatRoomId],
+          'messages',
+          [].concat(
+            (state.chatRoomUnread[chatRoomId].mentions || []).map(m => Object.assign({ type: MESSAGE_TYPES.TEXT }, m)),
+            (state.chatRoomUnread[chatRoomId].others || []).map(o => Object.assign({ type: MESSAGE_TYPES.INTERACTIVE }, o))
+          )
+        )
+      }
+      if (state.chatRoomUnread[chatRoomId].mentions) {
+        Vue.set(state.chatRoomUnread, chatRoomId, {
+          readUntil: { messageHash, createdDate, deletedDate: null },
+          mentions: [],
+          messages: state.chatRoomUnread[chatRoomId].messages
+            .filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
+        })
+        return
+      }
+    }
     Vue.set(state.chatRoomUnread, chatRoomId, {
       readUntil: { messageHash, createdDate, deletedDate: null },
-      mentions: prevMentions.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime()),
-      others: prevOthers.filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
+      messages: state.chatRoomUnread[chatRoomId].messages
+        .filter(m => new Date(m.createdDate).getTime() > new Date(createdDate).getTime())
     })
   },
   deleteChatRoomReadUntil (state, { chatRoomId, deletedDate }) {
     Vue.set(state.chatRoomUnread[chatRoomId].readUntil, 'deletedDate', deletedDate)
   },
-  addChatRoomUnreadMessage (state, { chatRoomId, messageHash, createdDate, isDMOrMention }) {
-    const newItem = { messageHash, createdDate }
-
-    if (isDMOrMention) {
-      state.chatRoomUnread[chatRoomId].mentions.push(newItem)
-    } else {
-      if (!('others' in state.chatRoomUnread[chatRoomId])) {
-        Vue.set(state.chatRoomUnread[chatRoomId], 'others', [newItem])
-      } else {
-        state.chatRoomUnread[chatRoomId].others.push(newItem)
-      }
-    }
+  addChatRoomUnreadMessage (state, { chatRoomId, messageHash, createdDate, type }) {
+    state.chatRoomUnread[chatRoomId].messages.push({ messageHash, createdDate, type })
   },
   deleteChatRoomUnreadMessage (state, { chatRoomId, messageHash }) {
-    const prevMentions = state.chatRoomUnread[chatRoomId].mentions
-    const prevOthers = state.chatRoomUnread[chatRoomId].others || []
+    // eslint-disable-next-line no-lone-blocks
+    {
+      // TODO: need to remove this block after we release 0.2.*
+      if (state.chatRoomUnread[chatRoomId].mentions) {
+        Vue.set(
+          state.chatRoomUnread[chatRoomId],
+          'mentions',
+          state.chatRoomUnread[chatRoomId].mentions.filter(m => m.messageHash !== messageHash)
+        )
+      }
+    }
```

Comment:
Again, this also doesn't make sense to me. Why is this here? `'state/vuex/postUpgradeVerification'` should have taken care of this and migrated everything to `messages`